keras/backend/cntk_backend.py

```diff
@@ -1576,6 +1587,14 @@ def __call__(self, inputs):
                value.dtype != np.float32 and
                value.dtype != np.float64):
                 value = value.astype(np.float32)
+            # in current version cntk can't support input with variable
+            # length. Will support it in next release.
+            if not self._is_input_shape_compatible(value, tensor):
+                raise ValueError('CNTK backend: The placeholder has been resolved '
+                                 'to shape `%s`, but input shape is `%s`. Currently '
+                                 'CNTK can not take variable length inputs. Please '
+                                 'padding the inputs before fit the data.'
```

Comment:
The recommendation (padding) seems to specialized to certain use cases (e.g. sequences). I would leave it out, and simply use "Please pass inputs that have a static shape."